pkg/operator/datasources.go

```diff
@@ -440,6 +443,62 @@ func (op *Reporting) handlePrestoTableDataSource(logger log.FieldLogger, dataSou
 	return nil
 }
 
+// TODO: investigate alternative names. exposeUnmanagedTable?
+func (op *Reporting) handleLinkExistingTable(logger log.FieldLogger, dataSource *metering.ReportDataSource) error {
+	if dataSource.Spec.LinkExistingTable.TableName == "" {
+		return fmt.Errorf("invalid configuration passed: spec.linkExistingTable.tableName field cannot be empty")
+	}
+	inputs := strings.Split(dataSource.Spec.LinkExistingTable.TableName, ".")
+	if len(inputs) != 3 {
```

Comment:
What is this magic number 3?

Reply:
We're hoping that the `spec.LinkExistingTable.TableName` in the form of a fully-qualified tables name: `catalog.schema.table_name`, and when that string is split by the `.` delimiter and should have three elements. That said, you could argue this needs a constant variable but I'm not sure how much extra value that offers.

Reply:
```suggestion
	if len(inputs) != ENOUGH_DOTS_FOR_FQTN {
```
Idk? Something like that, magic numbers slow me down reading the code, because they make me stop and make sure I understand what's going on.

Reply:
Updated to use a constant:
```go
	// given a table_name in the form of "catalog.schema.table_name", we
	// expect that splitting this overall string by the `.` delimiter
	// will yield an array of three string elements
	expectedArrSplitElementsFQTN = 3
```